front/src/modules/trainschedule/components/Timetable/Timetable.tsx

```diff
@@ -49,20 +62,21 @@ const Timetable = ({
   conflicts,
   upsertTrainSchedules,
   removeTrains,
-  setTrainIdToEdit,
-  trainIdToEdit,
+  setItemIdToEdit,
+  itemIdToEdit,
   trainSchedules = [],
   trainSchedulesWithDetails,
   dtoImport,
 }: TimetableProps) => {
   const { t } = useTranslation(['operationalStudies/scenario', 'common/itemTypes']);
+  const showPacedTrains = useSelector(getShowPacedTrains);
 
-  const [displayedTrainSchedules, setDisplayedTrainSchedules] = useState<
-    TrainScheduleWithDetails[]
-  >([]);
+  const [displayedTimetableItems, setDisplayedTimetableItems] = useState<TimetableItemResult[]>([]);
   const [conflictsListExpanded, setConflictsListExpanded] = useState(false);
-  const [selectedTrainIds, setSelectedTrainIds] = useState<TimetableItemId[]>([]);
+  const [selectedTimetableItemIdsByType, setSelectedTimetableItemIdsByType] =
+    useState<SelectedTimetableIdsByType>({ trainScheduleIds: [], pacedTrainIds: [] });
```

Comment:
Hum I have read all the code, and I think there is no gain in storing the timetableItems separately by type. It makes the code hardier to read, because you always need to test the item type before checking if it is selected or not.
Actually, we don't need this distinction, because the ids of the paced trains and of the trainSchedules can not be equals.

It would be better to store the selected ids in an simple array like this: 
```suggestion
  const [selectedTimetableItemIds, setSelectedTimetableItemIds] =
    useState<TimetableItemId[]>([]);
```
It will allow you to remove a lot of code

Reply:
After discussions, we decided to keep the current version for now since the proposition may bring some performance issues on big timetables.
Later, we might test the suggestion and measure the performances to decide whether or not we should change the code

Reply:
What are the performance issues?

Reply:
This was hypothetically. 
The question was about the count of each type (service, unique) separetly. 
If i keep 2 array, i can assign or unassign the train in/from the right one directly and keep the count of each array easily.
If i have just one array, i need to parse the entire array to count the two types, each time i select or unselect an item.

So we were wondering which method was the best in term of performance.

After some digging, I think that in our case it is not really useful to ask the question because we would have to exceed 100k entries before noticing notable differences

Reply:
Done